docs/extending/developing-code/code-overview/evaluation.rst

```diff
@@ -36,37 +38,35 @@ that, the method handle ``b`` needs to be computed. And this comes
 from a class heirarchy.
 
 Mathics and WL are not Object Oriented, so there is no such
-class-hierarchy lookup.  However, as mentioned above, pattern matching
+class-hierarchy lookup.  Instead, as mentioned above, pattern matching
 is used to decide which method of the object to call.
 
 Function Name to Python method lookup
--------------------------------------
+=====================================
 
 .. index:: Symbol, Predefined, Builtin, Expression
 
 When an ``Expression`` has not been rewritten, the entire function
 invocation in Mathics comes from the first leaf (or ``Head[]``) which
 should be a ``Symbol``. In Python this will be a class some sort, such
-as ``Builtin`` or ``Predefined`` or ``SympyFunction``. These classes
-are described in a later section.
+as ``Builtin`` or ``Predefined`` or ``SympyFunction`` or a method
```

Comment:
Also, it is possible to reduce to applying certain replacement rules stored in the Definitions object, which is the most typical case.